### ISSUE TITLE
[Calendar] - Error when using maxDate

### ISSUE THREAD

**Issue by angel1st** [syncfusion/flutter-examples#178]
Hi there,
Here is my issue - I have an app, where I use [bloclibrary](https://github.com/felangel/bloc/) along with syncfusion calendar.
When I don't use `maxData` the calendar works just fine with the BlocBuilder. However, at the moment, I added `maxDate`, the app enters in a vicious cycle and basically stops responding. 
I prepared a [minimum reproductive example](https://github.com/angel1st/flutter_calendar).
How to reproduce the issue:
1. Comment [line 92](https://github.com/angel1st/flutter_calendar/blob/master/lib/main.dart#L92) - build - everything works just fine
2. Uncomment line 92, build and look at the console - you will see the transitions coming from Bloc creates the cycle.

Let me know, shall you need any further inputs to fix the issue!
Thanks!

**Comment by Indumathi1195R** [comment]
Hi @angel1st ,

Based on your provided information we have checked the issue in the attached sample “**Endless loop occurred while using maxDate of the calendar with BlocLibrary**” and we have already found and fixed the mentioned issue. We will include this issue fix in our upcoming 2020 Volume 2 SP1 release, which is expected to roll out by mid of August 2020 and let you know once release rolled out. We appreciate your patience until then.

Also, it can fix through the sample level by assign the private variable value to the maxDate property of the SfCalendar. Please find the code snippet for the same.


class _MyHomePageState extends State<MyHomePage> {   CalendarController _calendarController;   DateTime _maxDate;   @override   void initState() {     super.initState();     _calendarController = CalendarController();     _maxDate=DateTime.now();   }body: BlocBuilder<HistoryCalendarBloc, HistoryCalendarState>(         condition: (_, historyCalendarState) =>             historyCalendarState is HistoryCalendarStateSet \|\|             historyCalendarState is HistoryCalendarStateIsLoading,         builder: (context, historyCalendarState) => LoadingOverlay(           isLoading: historyCalendarState is HistoryCalendarStateIsLoading,           child: SfCalendar(             controller: _calendarController,             maxDate: _maxDate,             //minDate: DateTime.now().subtract(const Duration(days: 30)),             firstDayOfWeek:                 MaterialLocalizations.of(context).firstDayOfWeekIndex,             view: CalendarView.month,             onViewChanged: (ViewChangedDetails viewChangedDetails) {               historyCalendarBloc.add(                 HistoryCalendarEventSetDateRange(                   startDate: viewChangedDetails.visibleDates.first,                   endDate: viewChangedDetails.visibleDates.last,   //                      .add(const Duration(days: 1)),                 ),               );             },           ),         ),       ),
--



Regards,
Indumathi R

**Comment by Indumathi1195R** [comment]
Hi Team,

Based on your provided information we have checked the issue in the attached sample “**Endless loop occurred while using maxDate of the calendar with BlocLibrary**” and we have already found and fixed the mentioned issue. We will include this issue fix in our upcoming 2020 Volume 2 SP1 release, which is expected to roll out by mid of August 2020 and let you know once release rolled out. We appreciate your patience until then.

Also, it can fix through sample level by assign the private variable value to the maxDate property of the SfCalendar. Please find the code snippet for the same.

class _MyHomePageState extends State<MyHomePage> {   CalendarController _calendarController;   DateTime _maxDate;   @override   void initState() {     super.initState();     _calendarController = CalendarController();     _maxDate=DateTime.now();   }body: BlocBuilder<HistoryCalendarBloc, HistoryCalendarState>(         condition: (_, historyCalendarState) =>             historyCalendarState is HistoryCalendarStateSet \|\|             historyCalendarState is HistoryCalendarStateIsLoading,         builder: (context, historyCalendarState) => LoadingOverlay(           isLoading: historyCalendarState is HistoryCalendarStateIsLoading,           child: SfCalendar(             controller: _calendarController,             maxDate: _maxDate,             //minDate: DateTime.now().subtract(const Duration(days: 30)),             firstDayOfWeek:                 MaterialLocalizations.of(context).firstDayOfWeekIndex,             view: CalendarView.month,             onViewChanged: (ViewChangedDetails viewChangedDetails) {               historyCalendarBloc.add(                 HistoryCalendarEventSetDateRange(                   startDate: viewChangedDetails.visibleDates.first,                   endDate: viewChangedDetails.visibleDates.last,   //                      .add(const Duration(days: 1)),                 ),               );             },           ),         ),       ),
--

Regards,
Indumathi R

**Comment by Indumathi1195R** [comment]
Hi Team,

Based on your provided information we have checked the issue in attached sample “Endless loop occurred while using maxDate of calendar with BlocLibrary” and we have already found and fixed the mentioned issue. We will include this issue fix in our upcoming 2020 Volume 2 SP1 release, which is expected to roll out by mid of August 2020 and let you know once release rolled out. We appreciate your patience until then.

Also, it can fix through sample level by assign the private variable value to the maxDate property of the SfCalendar. Please find the code snippet for the same.

Code snippet:

```
class _MyHomePageState extends State<MyHomePage> {   CalendarController _calendarController;   DateTime _maxDate;   @override   void initState() {     super.initState();     _calendarController = CalendarController();     _maxDate=DateTime.now();   }body: BlocBuilder<HistoryCalendarBloc, HistoryCalendarState>(         condition: (_, historyCalendarState) =>             historyCalendarState is HistoryCalendarStateSet \|\|             historyCalendarState is HistoryCalendarStateIsLoading,         builder: (context, historyCalendarState) => LoadingOverlay(           isLoading: historyCalendarState is HistoryCalendarStateIsLoading,           child: SfCalendar(             controller: _calendarController,             maxDate: _maxDate,             //minDate: DateTime.now().subtract(const Duration(days: 30)),             firstDayOfWeek:                 MaterialLocalizations.of(context).firstDayOfWeekIndex,             view: CalendarView.month,             onViewChanged: (ViewChangedDetails viewChangedDetails) {               historyCalendarBloc.add(                 HistoryCalendarEventSetDateRange(                   startDate: viewChangedDetails.visibleDates.first,                   endDate: viewChangedDetails.visibleDates.last,   //                      .add(const Duration(days: 1)),                 ),               );             },           ),         ),       ),
--
```

Regards,
Indumathi R

**Comment by angel1st** [comment]
Thanks, @Indumathi1195R!

**Comment by angel1st** [comment]
@JayavigneshwaranG - is the issue fixed in your August release?

**Comment by Indumathi1195R** [comment]
Hi @angel1st,

Yes, we have included the issue fix in our Weekly Nuget package version (**18.2.54**). Kindly update the calendar package to latest version for resolving the issue. Please find the latest versions from the following link.

**Link:** https://pub.dev/packages/syncfusion_flutter_calendar/versions

Regards,
Indumathi R
